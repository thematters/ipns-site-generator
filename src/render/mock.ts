import { ArticlePageContext, HomepageContext } from '../types'

const MOCK_AUTHOR = {
  userName: 'Anim',
  displayName: 'Excepteur Non',
}

const MOCK_ARTICLE = {
  author: MOCK_AUTHOR,
  title: 'Excepteur in consequat mollit consectetur.',
  description:
    'Aliquip reprehenderit elit nulla commodo sit. Aute deserunt quis cupidatat ea quis. Adipisicing magna tempor esse ea anim. Veniam aliquip laborum aliquip est laborum irure duis irure adipisicing dolore laboris in irure. Non sunt esse cillum consequat. Cupidatat qui ex fugiat ullamco cupidatat eu eu eu labore excepteur. Tempor tempor mollit nulla qui Lorem aute consectetur sint sint. Nostrud cupidatat ullamco ea elit in voluptate do mollit veniam.',
  summary: 'Ad cupidatat adipisicing consequat ad sint cillum dolore.',
  date: '2022-11-18T08:42:04.146Z',
  uri: './Qmc919SaZGj1yeDCLHx7KMY7WMzy1PF6UjixuapcG1bfAB',
  content: `
  <figure class="image">
    <img src="https://assets.matters.news/processed/1080w/profileCover/aa57a1ce-8926-4512-81d8-462f68fa3917.webp" data-asset-id="aa57a1ce-8926-4512-81d8-462f68fa3917">
    <figcaption><span>Reprehenderit proident sit consectetur id consequat officia.</span></figcaption>
  </figure>

  <h2>Duis ea voluptate cupidatat ad</h2>
  <p>Elit consequat labore tempor Lorem voluptate occaecat nostrud laborum minim. Cillum veniam ea cupidatat nulla commodo sunt amet magna amet sit culpa nulla deserunt <a href="https://google.com" target="_blank">reprehenderit</a> duis. Et nostrud sunt ad cupidatat laboris. Reprehenderit dolor dolore elit voluptate ex. Fugiat in in officia non eiusmod irure et. Velit ut aliquip ipsum exercitation exercitation nisi voluptate enim amet exercitation. Et consectetur ex nisi anim id consequat eiusmod veniam ipsum ullamco nulla deserunt nostrud.</p>

  <figure class="embed-video">
    <div class="iframe-container">
      <iframe src="https://www.youtube.com/embed/rokGy0huYEA?rel=0" frameborder="0" allowfullscreen="true" sandbox="allow-scripts allow-same-origin allow-popups">
      </iframe>
    </div>
    <figcaption><span>Nulla in fugiat labore ad.</span></figcaption>
  </figure>

  <h2>Cupidatat amet fugiat culpa id</h2>
  <p>Et ut dolore dolore ex. Deserunt adipisicing id dolor eiusmod minim ea. Pariatur veniam velit ad culpa nisi sit. Non nostrud irure nulla pariatur ipsum irure fugiat anim id Lorem duis. Ullamco incididunt ex ullamco elit. Amet voluptate minim laborum anim duis aliquip officia enim Lorem mollit aliquip laboris. <i>Mollit pariatur sunt pariatur occaecat deserunt esse</i>. Est eu ut elit id nisi duis id magna <strong>commodo ex et id sint laboris</strong>.</p>

  <blockquote>
    <strong>Enim aliqua est proident commodo dolor incididunt eiusmod. Anim anim eu pariatur aliqua qui. Sit non commodo enim ut aute officia eu. Adipisicing proident eu velit id proident.</strong>
  </blockquote>

  <p>Voluptate officia adipisicing voluptate amet dolore ad tempor aliquip reprehenderit Lorem. Ad dolor id minim occaecat ea non nulla. Ullamco exercitation consectetur duis tempor incididunt qui id. Sunt voluptate qui ex do Lorem consectetur laborum mollit culpa sunt anim occaecat esse. Velit deserunt eiusmod deserunt. Anim ullamco ad minim velit nulla aliquip culpa consequat laboris quis ad Lorem pariatur. Occaecat sunt irure reprehenderit.</p>

  <pre class="ql-syntax">query {\n&nbsp; article(\n&nbsp; &nbsp; input: { mediaHash: "zdpuAxP6uSfum74VS3pYmzBR9xvPbrBcX3J8BPpB3xdRGjVsX" }\n&nbsp; ) {\n&nbsp; &nbsp; id\n&nbsp; &nbsp; title\n&nbsp; &nbsp; summary\n&nbsp; }\n}\n</pre>

  <h2>Officia amet minim proident labore</h2>
  <h3>Proident fugiat amet</h3>
  <p>Duis eiusmod mollit ipsum exercitation voluptate sit ullamco.</p>
  <h3>Labore aute ea irure</h3>
  <ul>
    <li>
      <p>Adipisicing nisi deserunt velit proident nostrud et ipsum amet mollit.</p>
    </li>
    <li>
      <p>Esse nostrud deserunt Lorem pariatur incididunt.</p>
    </li>
    <li>
      <p>Non minim esse qui mollit consequat.</p>
      <ul>
        <li>
          <p>Exercitation dolor fugiat esse officia cupidatat anim.</p>
        </li>
        <li>
          <p>Esse eu anim irure voluptate non laborum laborum dolore dolore.</p>
        </li>
      </ul>
    </li>
    <li>
      <p>Laboris et excepteur est adipisicing magna qui do sit eiusmod.</p>
    </li>
  </ul>
  <h3>Qui aute voluptate</h3>
  <p>Labore dolor laboris anim. Laborum ut eiusmod et et minim duis aliquip deserunt laboris.</p>
  `,
}

export const MOCK_HOMEPAGE: HomepageContext = {
  meta: {
    title: `${MOCK_AUTHOR.displayName}'s homepage`,
    description:
      'Quis non enim sunt ut proident eu amet eiusmod do nulla ut laborum incididunt commodo officia.',
    authorName: `${MOCK_AUTHOR.displayName}`,
  },
  byline: {
    author: {
      name: MOCK_AUTHOR.displayName,
      uri: `https://matters.news/@${MOCK_AUTHOR.userName}`,
    },
    website: {
      name: 'Matters',
      uri: 'https://matters.news',
    },
  },
  rss: {
    ipns: 'k51qzi5uqu5dhihcpntbdym8g6desc80za48wshqs85no166wq3c8ee26cysot',
    xml: 'https://matters.news',
    json: 'https://matters.news',
  },
  articles: [
    MOCK_ARTICLE,
    MOCK_ARTICLE,
    MOCK_ARTICLE,
    MOCK_ARTICLE,
    MOCK_ARTICLE,
    MOCK_ARTICLE,
    MOCK_ARTICLE,
  ],
}

export const MOCK_ARTICLE_PAGE: ArticlePageContext = {
  encrypted: false,
  meta: {
    title: MOCK_ARTICLE.title,
    description: MOCK_ARTICLE.description,
    authorName: MOCK_ARTICLE.author.displayName,
    image:
      'https://assets.matters.news/processed/1080w/profileCover/aa57a1ce-8926-4512-81d8-462f68fa3917.webp',
  },
  byline: {
    date: '2022-11-18T08:42:04.146Z',
    author: {
      name: MOCK_AUTHOR.displayName,
      uri: `https://matters.news/@${MOCK_AUTHOR.userName}`,
    },
    website: {
      name: 'Matters',
      uri: 'https://matters.news',
    },
  },
  rss: {
    ipns: 'k51qzi5uqu5dhihcpntbdym8g6desc80za48wshqs85no166wq3c8ee26cysot',
    xml: 'https://matters.news',
    json: 'https://matters.news',
  },
  article: MOCK_ARTICLE,
}

export const MOCK_META_DATA = {
  contentHash: 'ipfs-hash',
  author: {
    name: 'Test User',
    url: 'user-home-page',
    description: 'this is a test user',
  },
  description: 'This is a piece of test content',
  image: 'image-url',
}
