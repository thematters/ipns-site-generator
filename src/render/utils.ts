export function stripSpaces(content: string) {
  return content?.replaceAll(/[\b\s]+/g, ' ').trim()
}

export function addLeadingZero(num: number, len: number = 2) {
  return `00${num}`.slice(-len)
}

export function shortDate(date: string) {
  const d = new Date(date)
  const dayNumber = d.getDate()
  const month = d.getMonth() + 1 // from range 0-11 to 1-12
  const year = d.getFullYear()

  return `${year}-${addLeadingZero(month)}-${addLeadingZero(dayNumber)}`
}
