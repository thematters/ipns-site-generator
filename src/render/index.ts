import path from 'path'
import nunjucks from 'nunjucks'
import { ArticlePageContext, HomepageContext } from '../types'

import { shortDate, stripSpaces } from './utils'

// configure nunjucks
nunjucks
  .configure(path.resolve(__dirname, 'views'))
  .addFilter('shortDate', shortDate)
  .addFilter('stripSpaces', stripSpaces)

// render templates
export const renderArticlePage = (context: ArticlePageContext) => {
  return nunjucks.render('article/index.njk', context)
}

export const renderHomepage = (context: HomepageContext) => {
  return nunjucks.render('homepage/index.njk', context)
}
