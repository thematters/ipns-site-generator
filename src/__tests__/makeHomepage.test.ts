import fetch from 'isomorphic-fetch'
import {
  makeHomepage,
  makeHomepageBundles,
  makeActivityPubBundles,
} from '../makeHomepage'
import { MOCK_HOMEPAGE } from '../render/mock'

jest.mock('isomorphic-fetch')
const mockedFetch = fetch as jest.Mock

describe('makeHomepage', () => {
  test('can generate HTML, XML and JSON of homepage', async () => {
    mockedFetch.mockResolvedValue({
      arrayBuffer: () => Promise.resolve(new ArrayBuffer(1)),
    })

    const { html, xml, json } = await makeHomepage(
      MOCK_HOMEPAGE('matters.news')
    )
    expect(html).toMatchSnapshot()
    expect(xml).toMatchSnapshot()
    expect(json).toMatchSnapshot()
  })

  test('can generate bundles of HTML, XML and JSON', async () => {
    mockedFetch.mockResolvedValue({
      arrayBuffer: () => Promise.resolve(new ArrayBuffer(1)),
    })

    const bundles = await makeHomepageBundles(MOCK_HOMEPAGE('matters.news'))

    let html = ''
    let xml = ''
    let json = ''
    for (const { path, content } of bundles) {
      switch (path) {
        case 'index.html':
          html = content
          break
        case 'rss.xml':
          xml = content
          break
        case 'feed.json':
          json = content
          break
      }
    }
    expect(html).toMatchSnapshot()
    expect(xml).toMatchSnapshot()
    expect(json).toMatchSnapshot()
  })

  test('can generate activitypub file bundles of webfinger, outbox, etc.', async () => {
    mockedFetch.mockResolvedValue({
      arrayBuffer: () => Promise.resolve(new ArrayBuffer(1)),
    })

    const bundles = await makeActivityPubBundles(MOCK_HOMEPAGE('matters.news'))

    let webfinger: string = ''
    for (const { path, content } of bundles) {
      switch (path) {
        case '.well-known/webfinger':
          webfinger = content
          break
      }
    }
    expect(webfinger).toMatchSnapshot()
  })
})
