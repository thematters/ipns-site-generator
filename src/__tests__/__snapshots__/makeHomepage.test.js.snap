// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeHomepage can generate basic HTML bundle 1`] = `
"

<html>
  <head>
    <meta charset=\\"UTF-8\\" />
<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
<title>Excepteur Non&#39;s homepage</title>
<meta name=\\"description\\" content=\\"Quis non enim sunt ut proident eu amet eiusmod do nulla ut laborum incididunt commodo officia.\\">
<meta property=\\"og:title\\" content=\\"Excepteur Non&#39;s homepage\\">
<meta property=\\"og:description\\" content=\\"Quis non enim sunt ut proident eu amet eiusmod do nulla ut laborum incididunt commodo officia.\\">
<meta property=\\"og:image\\" content=\\"{ {meta.image }}\\" />
<meta property=\\"article:author\\" content=\\"Excepteur Non\\">
<meta name=\\"twitter:title\\" content=\\"Excepteur Non&#39;s homepage\\">
<meta name=\\"twitter:description\\" content=\\"Quis non enim sunt ut proident eu amet eiusmod do nulla ut laborum incididunt commodo officia.\\">
<meta name=\\"twitter:image\\" content=\\"\\" />
<link rel=\\"alternate\\" type=\\"application/atom+xml\\" href=\\"https://matters.news\\" />
<link rel=\\"alternate\\" type=\\"application/json\\" href\\"https://matters.news\\" />


    
<style>
:root {
  /* typography: font-size, font-weight, line-height, font-family */
  --font-size-x-small: 11px;
  --font-size-small: 13px;
  --font-size-base: 16px;
  --font-size-medium: 18px;
  --font-size-large: 24px;
  --font-size-x-large: 32px;

  --line-height-base: 1.5;

  --font-weight-base: 400;
  --font-weight-medium: 500;

  --font-family: -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", sans-serif;

  /* colors */
  --color-grey-lighter: #f7f7f7;
  --color-grey-light: #DDDDDD;
  --color-grey: #B3B3B3;
  --color-grey-darker: #808080;
  --color-black: #000000;
  --color-white: #fff;

  /* spacings */
  --spacing-xx-small: 8px;
  --spacing-x-small: 8px;
  --spacing-small: 12px;
  --spacing-base: 16px;
  --spacing-medium: 24px;
  --spacing-large: 32px;
  --spacing-x-large: 40px;
  --spacing-xx-large: 48px;

}
</style>

<style>
/* reset */
/* Document
   ========================================================================== */
html {
  box-sizing: border-box;
  border-collapse: collapse;
  -webkit-text-size-adjust: 100%;
}

html,
body {
  padding: 0;
  margin: 0;
}

* {
  border-collapse: inherit;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

/* Sections
   ========================================================================== */
main {
  display: block;
}

ol,
ul {
  list-style: none;
}

ol,
ul,
li {
  padding: 0;
  margin: 0;
  border: 0;
}

/* Text-level semantics
   ========================================================================== */
h1,
h2,
h3,
h4,
h5,
h6,
p,
blockquote,
pre,
code {
  padding: 0;
  margin: 0;
  border: 0;
}

code,
kbd,
samp,
pre {
  font-family: monospace;
  font-size: 1em;
}

pre {
  white-space: pre-wrap;
}

a {
  background-color: transparent;
}

b,
strong {
  font-weight: bold;
}

small {
  font-size: 80%;
}

/* Embedded content
   ========================================================================== */
img {
  border-style: none;
}

/* Forms
   ========================================================================== */
form {
  width: 100%;
}

input,
select,
textarea,
button {
  padding: 0;
  margin: 0;
  font: inherit;
  line-height: 1.15;
  color: inherit;
  border: 0;
  border-radius: 0;
  outline: 0;
  appearance: none;
}

button {
  display: inline-block;
  width: auto;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  user-select: none;
  background: transparent;
}

button > * {
  vertical-align: middle;
}

button::-moz-focus-inner {
  padding: 0;
  border: 0;
}

button:disabled,
button[aria-disabled='true'] {
  cursor: default;
  cursor: not-allowed;
}

button,
input {
  overflow: visible;
}

button,
select {
  text-transform: none;
}

input,
select,
textarea {
  display: block;
  width: 100%;
}

/* base */
body {
  font-family: var(--font-family);
  font-size: var(--font-size-base);
  line-height: var(--line-height-base);
  background: var(--background-color);
  color: var(--text-color);
}

a {
  &:link {
    color: #0070E0;
    text-decoration: none;
  }

  &:visited,
  &:hover,
  &:active,
  &:focus {
    color: #7D1BFA;
    text-decoration: none;
  }
}

hr { height: 1px; }

h1, h2, h3, h4, h5, h6 {
  font-weight: var(--font-weight-medium);
  line-height: 1.4;
}
h1 { font-size: var(--font-size-x-large); }
h2 { font-size: var(--font-size-large); }
h3 { font-size: var(--font-size-medium); }
h4 { font-size: var(--font-size-base); }
h5 { font-size: var(--font-size-small); }
h6 { font-size: var(--font-size-x-small); }

li ul, li ol { margin: 0 var(--spacing-base); }
li { margin: var(--spacing-base) 0; }
ul { list-style-type: disc; }
ol { list-style-type: decimal; }
ol ol { list-style: upper-alpha; }
ol ol ol { list-style: lower-roman; }
ol ol ol ol { list-style: lower-alpha; }

img, video, audio {
  display: block;
  max-width: 100%;
  margin: 0 auto;
}
audio {
  width: 100%;
}

.copied {
  color: var(--color-grey);
}
[data-clipboard] {
  cursor: pointer;
}
</style>

<style>
main {
  max-width: 673px;
  width: 100%;
  padding: 0 var(--spacing-base);
  margin: 0 auto;
}
</style>

<style>
/* light mode */
:root {
  --text-color: var(--color-black);
  --background-color: var(--color-white);
  --date-color: var(--color-grey);
  --border-color: var(--color-grey-light);
  --code-background-color: var(--color-grey-lighter);
}

/* dark mode */
:root.dark {
  --text-color: var(--color-white);
  --background-color: var(--color-black);
  --date-color: var(--color-grey);
  --border-color: var(--color-grey-darker);
  --code-background-color: var(--color-grey-darker);
}
</style>



<style>
.themeSwitch {
  display: inline-flex;
}

.themeSwitch svg {
  width: 20px;
  height: 20px;
}

/* light mode */
:root:not(.dark) .themeSwitch .dark {
  display: none;
}

/* dark mode */
:root.dark .themeSwitch .light {
  display: none;
}
</style>

<style>
.byline {
  margin-top: var(--spacing-base);
  margin-bottom: var(--spacing-medium);
  font-size: var(--font-size-base);
  font-weight: var(--font-weight-medium);
  line-height: 1;
}

.byline time {
  color: var(--date-color);
}

@media(min-width: 706px) {
  .byline {
    margin-bottom: var(--spacing-x-large);
  }
}
</style>

<style>
.rssEntrances {
  font-size: var(--font-size-base);
  font-weight: var(--font-weight-medium);
  line-height: 1;
}

.rssEntrances ul,
.rssEntrances li {
  padding: 0;
  margin: 0;
  list-style-type: none;
}

.rssEntrances ul {
  display: inline-flex;
  justify-content: space-between;
  align-items: center;
}

.rssEntrances li {
  display: inline-flex;
  align-items: center;
  padding: 0 var(--spacing-x-small);
}

.rssEntrances li:first-child {
  padding-left: 0;
}

.rssEntrances li + li {
  border-left: 2px solid var(--border-color);
}

.rssEntrances a {
  color: inherit;
  text-decoration: none;
  cursor: pointer;
}

.rssEntrances svg {
  margin-right: var(--spacing-xx-small);
  color: var(--border-color);
}
</style>


    <style>

header nav {
  margin-top: var(--spacing-x-large);
  display: flex;
  justify-content: space-between;
  align-items: center;
}

header h1 {
  margin-top: var(--spacing-medium);
}


ul, li, ol {
  list-style-type: none;
}

ul > li {
  margin-bottom: var(--spacing-large);
}


.articleDigest h2 {
  line-height: 1.5;
}
.articleDigest p {
  margin-top: var(--spacing-small);
}
.articleDigest time {
  display: block;
  margin-top: var(--spacing-x-small);
  font-size: var(--font-size-small);
  color: var(--date-color);
  line-height: 1.23;
}
</style>

  </head>

  <body>
    <main>
      <header>
  <nav>
    <section class=\\"rssEntrances\\">
  <ul>
    <li>
      <svg width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<path d=\\"M3.99995 2.40002C3.78778 2.40002 3.5843 2.48431 3.43427 2.63434C3.28424 2.78437 3.19995 2.98785 3.19995 3.20002C3.19995 3.4122 3.28424 3.61568 3.43427 3.76571C3.5843 3.91574 3.78778 4.00002 3.99995 4.00002C8.41835 4.00002 12 7.58162 12 12C12 12.2122 12.0842 12.4157 12.2343 12.5657C12.3843 12.7157 12.5878 12.8 12.8 12.8C13.0121 12.8 13.2156 12.7157 13.3656 12.5657C13.5157 12.4157 13.6 12.2122 13.6 12C13.6 6.69842 9.30155 2.40002 3.99995 2.40002Z\\" fill=\\"currentColor\\"/>
<path d=\\"M3.1999 7.20002C3.1999 6.98785 3.28419 6.78437 3.43422 6.63434C3.58425 6.48431 3.78773 6.40002 3.9999 6.40002C5.48511 6.40002 6.9095 6.99002 7.9597 8.04023C9.0099 9.09043 9.5999 10.5148 9.5999 12C9.5999 12.2122 9.51562 12.4157 9.36559 12.5657C9.21556 12.7157 9.01208 12.8 8.7999 12.8C8.58773 12.8 8.38425 12.7157 8.23422 12.5657C8.08419 12.4157 7.9999 12.2122 7.9999 12C7.9999 10.9392 7.57847 9.92174 6.82833 9.1716C6.07818 8.42145 5.06077 8.00002 3.9999 8.00002C3.78773 8.00002 3.58425 7.91574 3.43422 7.76571C3.28419 7.61568 3.1999 7.4122 3.1999 7.20002ZM2.3999 12C2.3999 11.5757 2.56847 11.1687 2.86853 10.8687C3.16859 10.5686 3.57556 10.4 3.9999 10.4C4.42425 10.4 4.83122 10.5686 5.13127 10.8687C5.43133 11.1687 5.5999 11.5757 5.5999 12C5.5999 12.4244 5.43133 12.8313 5.13127 13.1314C4.83122 13.4315 4.42425 13.6 3.9999 13.6C3.57556 13.6 3.16859 13.4315 2.86853 13.1314C2.56847 12.8313 2.3999 12.4244 2.3999 12Z\\" fill=\\"currentColor\\"/>
</svg>

      <span aira-role=\\"button\\" aria-label=\\"IPNS address\\" data-clipboard=\\"k51qzi5uqu5dhihcpntbdym8g6desc80za48wshqs85no166wq3c8ee26cysot\\">IPNS</span>
    </li>
    <li>
      <a href=\\"https://matters.news\\" target=\\"_blank\\" aria-label=\\"subscribe RSS feed\\">RSS</a>
    </li>
    <li>
      <a href=\\"https://matters.news\\" target=\\"_blank\\" aria-label=\\"subscribe JSON feed\\">JSON</a>
    </li>
  </ul>
</section>

    <section class=\\"themeSwitch\\">
  <button class=\\"light\\" type=\\"button\\" aira-label=\\"switch to dark mode\\">
    <svg width=\\"20\\" height=\\"20\\" viewBox=\\"0 0 20 20\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<circle cx=\\"10.5\\" cy=\\"10\\" r=\\"7.375\\" stroke=\\"#333333\\" stroke-width=\\"1.25\\"/>
<path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M10.4646 5.2108C10.3559 5.20365 10.2462 5.20001 10.1357 5.20001C7.41112 5.20001 5.20239 7.40874 5.20239 10.1333C5.20239 12.858 7.41112 15.0667 10.1357 15.0667C10.2462 15.0667 10.3559 15.063 10.4646 15.0559V5.2108Z\\" fill=\\"#333333\\"/>
</svg>

  </button>

  <button class=\\"dark\\" type=\\"button\\" aira-label=\\"switch to light mode\\">
    <svg width=\\"20\\" height=\\"20\\" viewBox=\\"0 0 20 20\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <circle cx=\\"10\\" cy=\\"10\\" r=\\"7.5\\" stroke=\\"white\\"/>
  <path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M9.2356 15.0507C9.36756 15.0613 9.50099 15.0667 9.63568 15.0667C12.3603 15.0667 14.569 12.858 14.569 10.1333C14.569 7.40874 12.3603 5.20001 9.63568 5.20001C9.50099 5.20001 9.36756 5.20541 9.2356 5.216V15.0507Z\\" fill=\\"white\\"/>
</svg>

  </button>
</section>

  </nav>

  <h1 itemprop=\\"headline\\">Excepteur Non&#39;s homepage</h1>

  <section class=\\"byline\\" itemprops=\\"provider\\" itemscope itemtype=\\"http://schema.org/Organization\\">
  <a href=\\"https://matters.news/@Anim\\" target=\\"_blank\\">
    Excepteur Non
  </a>
  &nbsp;

  

  &nbsp;from&nbsp;
  <a href=\\"https://matters.news\\" target=\\"_blank\\">
    Matters
  </a>
</section>

</header>


      <ul>
      
        
<li>
  <section class=\\"articleDigest\\">
    <h2><a href=\\"./Qmc919SaZGj1yeDCLHx7KMY7WMzy1PF6UjixuapcG1bfAB\\">Excepteur in consequat mollit consectetur.</a></h2>
    <p>Aliquip reprehenderit elit nulla commodo sit. Aute deserunt quis cupidatat ea quis. Adipisicing magna tempor esse ea anim. Veniam aliquip laborum aliquip est laborum irure duis irure adipisicing dolore laboris in irure. Non sunt esse cillum consequat. Cupidatat qui ex fugiat ullamco cupidatat eu eu eu labore excepteur. Tempor tempor mollit nulla qui Lorem aute consectetur sint sint. Nostrud cupidatat ullamco ea elit in voluptate do mollit veniam.</p>
    <time itemprop=\\"datePublished\\" datetime=\\"2022-11-18T08:42:04.146Z\\">2022-11-18</time>
  </section>
</li>

      
        
<li>
  <section class=\\"articleDigest\\">
    <h2><a href=\\"./Qmc919SaZGj1yeDCLHx7KMY7WMzy1PF6UjixuapcG1bfAB\\">Excepteur in consequat mollit consectetur.</a></h2>
    <p>Aliquip reprehenderit elit nulla commodo sit. Aute deserunt quis cupidatat ea quis. Adipisicing magna tempor esse ea anim. Veniam aliquip laborum aliquip est laborum irure duis irure adipisicing dolore laboris in irure. Non sunt esse cillum consequat. Cupidatat qui ex fugiat ullamco cupidatat eu eu eu labore excepteur. Tempor tempor mollit nulla qui Lorem aute consectetur sint sint. Nostrud cupidatat ullamco ea elit in voluptate do mollit veniam.</p>
    <time itemprop=\\"datePublished\\" datetime=\\"2022-11-18T08:42:04.146Z\\">2022-11-18</time>
  </section>
</li>

      
        
<li>
  <section class=\\"articleDigest\\">
    <h2><a href=\\"./Qmc919SaZGj1yeDCLHx7KMY7WMzy1PF6UjixuapcG1bfAB\\">Excepteur in consequat mollit consectetur.</a></h2>
    <p>Aliquip reprehenderit elit nulla commodo sit. Aute deserunt quis cupidatat ea quis. Adipisicing magna tempor esse ea anim. Veniam aliquip laborum aliquip est laborum irure duis irure adipisicing dolore laboris in irure. Non sunt esse cillum consequat. Cupidatat qui ex fugiat ullamco cupidatat eu eu eu labore excepteur. Tempor tempor mollit nulla qui Lorem aute consectetur sint sint. Nostrud cupidatat ullamco ea elit in voluptate do mollit veniam.</p>
    <time itemprop=\\"datePublished\\" datetime=\\"2022-11-18T08:42:04.146Z\\">2022-11-18</time>
  </section>
</li>

      
        
<li>
  <section class=\\"articleDigest\\">
    <h2><a href=\\"./Qmc919SaZGj1yeDCLHx7KMY7WMzy1PF6UjixuapcG1bfAB\\">Excepteur in consequat mollit consectetur.</a></h2>
    <p>Aliquip reprehenderit elit nulla commodo sit. Aute deserunt quis cupidatat ea quis. Adipisicing magna tempor esse ea anim. Veniam aliquip laborum aliquip est laborum irure duis irure adipisicing dolore laboris in irure. Non sunt esse cillum consequat. Cupidatat qui ex fugiat ullamco cupidatat eu eu eu labore excepteur. Tempor tempor mollit nulla qui Lorem aute consectetur sint sint. Nostrud cupidatat ullamco ea elit in voluptate do mollit veniam.</p>
    <time itemprop=\\"datePublished\\" datetime=\\"2022-11-18T08:42:04.146Z\\">2022-11-18</time>
  </section>
</li>

      
        
<li>
  <section class=\\"articleDigest\\">
    <h2><a href=\\"./Qmc919SaZGj1yeDCLHx7KMY7WMzy1PF6UjixuapcG1bfAB\\">Excepteur in consequat mollit consectetur.</a></h2>
    <p>Aliquip reprehenderit elit nulla commodo sit. Aute deserunt quis cupidatat ea quis. Adipisicing magna tempor esse ea anim. Veniam aliquip laborum aliquip est laborum irure duis irure adipisicing dolore laboris in irure. Non sunt esse cillum consequat. Cupidatat qui ex fugiat ullamco cupidatat eu eu eu labore excepteur. Tempor tempor mollit nulla qui Lorem aute consectetur sint sint. Nostrud cupidatat ullamco ea elit in voluptate do mollit veniam.</p>
    <time itemprop=\\"datePublished\\" datetime=\\"2022-11-18T08:42:04.146Z\\">2022-11-18</time>
  </section>
</li>

      
        
<li>
  <section class=\\"articleDigest\\">
    <h2><a href=\\"./Qmc919SaZGj1yeDCLHx7KMY7WMzy1PF6UjixuapcG1bfAB\\">Excepteur in consequat mollit consectetur.</a></h2>
    <p>Aliquip reprehenderit elit nulla commodo sit. Aute deserunt quis cupidatat ea quis. Adipisicing magna tempor esse ea anim. Veniam aliquip laborum aliquip est laborum irure duis irure adipisicing dolore laboris in irure. Non sunt esse cillum consequat. Cupidatat qui ex fugiat ullamco cupidatat eu eu eu labore excepteur. Tempor tempor mollit nulla qui Lorem aute consectetur sint sint. Nostrud cupidatat ullamco ea elit in voluptate do mollit veniam.</p>
    <time itemprop=\\"datePublished\\" datetime=\\"2022-11-18T08:42:04.146Z\\">2022-11-18</time>
  </section>
</li>

      
        
<li>
  <section class=\\"articleDigest\\">
    <h2><a href=\\"./Qmc919SaZGj1yeDCLHx7KMY7WMzy1PF6UjixuapcG1bfAB\\">Excepteur in consequat mollit consectetur.</a></h2>
    <p>Aliquip reprehenderit elit nulla commodo sit. Aute deserunt quis cupidatat ea quis. Adipisicing magna tempor esse ea anim. Veniam aliquip laborum aliquip est laborum irure duis irure adipisicing dolore laboris in irure. Non sunt esse cillum consequat. Cupidatat qui ex fugiat ullamco cupidatat eu eu eu labore excepteur. Tempor tempor mollit nulla qui Lorem aute consectetur sint sint. Nostrud cupidatat ullamco ea elit in voluptate do mollit veniam.</p>
    <time itemprop=\\"datePublished\\" datetime=\\"2022-11-18T08:42:04.146Z\\">2022-11-18</time>
  </section>
</li>

      
      </ul>
    </main>

    <script type=\\"text/javascript\\">
function toggleDarkMode(on) {
  var $root = document.querySelector(\\":root\\");

  if (on) {
    $root.classList.add(\\"dark\\");
  } else {
    $root.classList.remove(\\"dark\\");
  }
}

// turn off
document.querySelector(\\".themeSwitch .dark\\").addEventListener(\\"click\\", () => {
  toggleDarkMode(false);
});

// turn on
document.querySelector(\\".themeSwitch .light\\").addEventListener(\\"click\\", () => {
  toggleDarkMode(true);
});
</script>

    <script type=\\"text/javascript\\">
var $copyButtons = document.querySelectorAll('[data-clipboard]');

for (var i = 0, len = $copyButtons.length; i < len; i++) {
  $copyButtons[i].addEventListener('click', onCopyButtonClick, false);
}

function onCopyButtonClick(event) {
  var target = event.target;
  var text = target.dataset.clipboard;
  copyTextToClipboard(text, target);
}

function copyTextToClipboard(text, $el) {
  var originalInnerText = $el.innerText;

  try {
    navigator.clipboard.writeText(text).then(function() {
      $el.classList.add('copied');
      $el.innerText = 'Copied';
      setTimeout(function() {
        $el.classList.remove('copied');
        $el.innerText = originalInnerText;
      }, 3000);
    }, function(err) {
      alert(text);
    });
  } catch (error) {
    alert(text);
  }
}
</script>

    <script async src=\\"https://www.googletagmanager.com/gtag/js?id=G-K4KK55LL24\\"></script>
<script>window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date); gtag('config', 'G-K4KK55LL24');</script>

  </body>
</html>
"
`;
